src/main/java/frc/trigon/robot/utilities/ShootingCalculations.java

```diff
@@ -0,0 +1,120 @@
+                new Translation2d(waypointsFromDistance[0].distance, waypointsFromDistance[0].velocity),
+                new Translation2d(waypointsFromDistance[1].distance, waypointsFromDistance[1].velocity),
+                distance
+        );
+        double angle = calculateBetweenTranslations(
+                new Translation2d(waypointsFromDistance[0].distance, waypointsFromDistance[0].angle),
+                new Translation2d(waypointsFromDistance[1].distance, waypointsFromDistance[1].angle),
+                distance
+        );
+        return new Waypoint(distance, velocity, angle);
+    }
+
+    /**
+     * Gets the velocity needed for the shooter to shoot from a distance.
+     *
+     * @param distance the distance from the target
+     * @return the velocity needed for the shooter
+     */
+    public static double getShootingVelocityFromDistance(double distance) {
+        return Objects.requireNonNullElse(getWaypointFromDistance(distance), new Waypoint(0, 0, 0)).velocity;
+    }
+
+    /**
+     * Gets the angle needed for the shooter to shoot from a distance.
+     *
+     * @param distance The distance from the target
+     * @return the angle needed for the shooter
+     */
+    public static double getShootingAngleFromDistance(double distance) {
+        return Objects.requireNonNullElse(getWaypointFromDistance(distance), new Waypoint(0, 0, 0)).angle;
+    }
+
+    /**
+     * Calculates the value between two translations.
+     *
+     * @param firstPoint  the first point
+     * @param secondPoint the second point
+     * @param pointToKnow the point to know the value of (between the first and second points)
+     * @return the value between the two points
+     */
+    public static double calculateBetweenTranslations(
+            Translation2d firstPoint, Translation2d secondPoint, double pointToKnow) {
+        double m = (firstPoint.getY() - secondPoint.getY()) / (firstPoint.getX() - secondPoint.getX());
+        return m * pointToKnow - m * firstPoint.getX() + firstPoint.getY();
+    }
+
+    /**
+     * Adds a waypoint to the waypoints list.
+     *
+     * @param distance the distance for the waypoint
+     * @param velocity the velocity for the waypoint
+     * @param angle    the angle for the waypoint
+     */
+    public static void addWaypoint(double distance, double velocity, double angle) {
+        Waypoint toAdd = new Waypoint(distance, velocity, angle);
+        waypoints.add(toAdd);
+    }
+
+    /**
+     * @return the waypoints list
+     */
+    public static ArrayList<Waypoint> getWaypoints() {
+        return waypoints;
+    }
+
+    /**
+     * Sets the waypoints list.
+     *
+     * @param waypoints the waypoints list
+     */
+    public static void setWaypoints(ArrayList<Waypoint> waypoints) {
+        ShootingCalculations.waypoints.clear();
+        ShootingCalculations.waypoints.addAll(waypoints);
+    }
+
+    private static Waypoint[] getNearstWaypointsFromDistance(double distance) {
```

Comment:
I want this to return the first two waypoints if the given distance is smaller than the smallest waypoint 
and return the last two waypoints if the given distance is further than the further waypoint
Only do this if you understand why